### PR TITLE
BloggerApp: Integration specs

### PR DESCRIPTION
**Pull request by BenjaminSemah**
### In this branch, we have done the following;
✅ used capybara to write integration test for the following views
 `Login page`
`User index page`
`User show page`
`User post index page`
`Post show page`
✅ fixed linter errors

### CODE REVIEW

File: spec/views/post_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      end
+    end
+
+    it 'displays content of each comment' do
+      expect(page).to have_content('My first comment')
+      expect(page).to have_content('My second comment')
+      expect(page).to have_content('My third comment')
+    end
+  end
+end
```
Comment (Emmyn5600) [inline spec/views/post_show_spec.rb:73]: - [ ] All the tests are failing because you have issues in the database, kindly fix them.
 
 
image


image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @BenjaminSemah  and @Lichapa 

Good job so far! I know you have worked hard on making this milestone. There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Required Changes :recycle:
- Check the comment under the review

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BenjaminSemah** [comment]
### [Loom Video Demonstration of  All Tests Passed](https://www.loom.com/share/438ea04f63d142caaa3fc57c96dc106c)

Hi.

The previous reviewer informed me that _*all the tests are failing*_ and that I should fix them.

The test are actually passing - all of them. So I have included a [link to a loom video](https://www.loom.com/share/438ea04f63d142caaa3fc57c96dc106c) demonstrating that all of my test are passing.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @BenjaminSemah, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] The issue pointed out by the previous reviewer usually occur when you delete migration files wrongly. I have seen another issue in the project models, you should explicitly tell the post table(model) that it `belongs_to :author, class_name: 'User'` and in user's table(model) `has_many :posts, foreign_key: :author_id`. Do this even in other models like comments and likes. Your tests can pass because you are testing views and you have the project with you, if you try to delete the project and clone from GitHub, I'm sure you will get the same errors because you will be required to run the migration files in order to have a running project
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BenjaminSemah** [comment]
### Code Review Suggestions Implemented

Hello. I have implemented the required changes pointed out  by the previous reviewers. 

I had another Micronaut run `rails db:migrate` on their end and it was successful as shown in the [screenshot](https://user-images.githubusercontent.com/88531553/169398403-61cbe5c1-caeb-4f7e-92a8-0c8157d833e6.PNG)

All of the 47 tests are now passing.

migrationfilepic

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**